src/axom/mir/MIRMesh.hpp

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * \file MIRMesh.hpp
+ * 
+ * \brief Contains the specification for the MIRMesh class.
+ * 
+ */
+
+#ifndef __MIR_MESH_H__
+#define __MIR_MESH_H__
+
+
+#include "axom/core.hpp"  // for axom macros
+#include "axom/slam.hpp"  // unified header for slam classes and functions
+
+#include "MIRMeshTypes.hpp"
+#include "CellData.hpp"
+
+// C/C++ includes
+#include <cmath>          // for definition of M_PI, exp()
+#include <vector>
+#include <string>
+#include <fstream>
+
+namespace numerics = axom::numerics;
+namespace slam = axom::slam;
+
+//--------------------------------------------------------------------------------
+namespace axom
+{
+namespace mir
+{
+
+  const int NULL_MAT = -1;
+
+  //--------------------------------------------------------------------------------
+
+  /**
+   * \class MIRMesh
+   * 
+   * \brief The MIRMesh class represents a finite element mesh containing element volume fractions.
+   * 
+   * \detail This class is meant to be used in conjunction with the InterfaceReconstructor class
+   *         to process the mesh.
+   * 
+   */
+  class MIRMesh
+  {
+    public:
+      /**
+       * \brief Default constructor.
+       */
+      MIRMesh();
+
+      /**
+       * \brief Copy constrcutor.
+       */
+      MIRMesh(MIRMesh* _mesh);
+
+      /**
+       * \brief Default destructor.
+       */
+      ~MIRMesh();
+
+      /**
+       * \brief Initializes a mesh with the provided data and topology.
+       * 
+       * \param _verts  The set of vertices of the mesh.
+       * \param _elems  The set of elements of the mesh.
+       * \param _numMaterials  The number of materials present in the mesh.
+       * \param _topology  The topology/connectivity of the mesh.
+       * \param _mapData  The data used to initialized the maps associated with the vertex and element sets.
+       * \param _elementVF  The volume fractions of each element. Note that this is an optional parameter.
+       */
+      void  initializeMesh(VertSet _verts, ElemSet _elems, int _numMaterials, CellTopologyData _topology, CellMapData _mapData, std::vector<std::vector<axom::float64> > _elementVF = {});
```

Comment:
```cpp
 std::vector<std::vector<axom::float64> > _elementVF
```

Please make sure that larger/more complex objects are passed by (const) reference.